include/rdma/fi_ext.h

```diff
@@ -203,6 +260,20 @@ struct fi_peer_srx_context {
 	struct fid_peer_srx *srx;
 };
 
+
+/*
+ * Peer transfers
+ */
+struct fi_peer_transfer_context {
+	size_t size;
+	struct fi_info *info;
+	struct fid_domain *domain;
```

Comment:
The domain isn't needed.  We already need to have a domain opened before opening the EP.  So, adding a peer domain would cover this.  What the offload provider needs is access to memory registration calls.